### PR TITLE
Feat/uneco coinjoin

### LINKED ISSUE #7377: Warn about uneconomical coinjoin
In Wasabi wallet, when the overall account balance is below 0.01 BTC (1 million sats) the application warns the user that "_Coinjoining might be uneconomical. Receive more funds or press play to bypass._"
- Let's replicate this warning in Suite when the user clicks the Anonymize button. Wasabi refers to the threshold value as `PlebStopThreshold`, but maybe we want to call it something like `UneconomicalCoinjoinThreshold`.
- Inform the user during coinjoin account onboarding that "_Coinjoin might be uneconomical if your account balance is below_ `UneconomicalCoinjoinThreshold`."

@hynek-jina please provide the design.

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* add an uneco threshhold to Suite. Did't find the `PlebStopThreshold `constant you mentioned, @andrewkozlik, only `plebsDontPayThreshold` related to fees. Should it also be renamed? Not sure having the word `plebs` in our codebase is nice, even if it's Wasabi's naming.
* Redo the CoinJoin warning in the receive section
* Add an uneco warning to the StatusWheel.

@hynek-jina message for the `TR_COINJOIN_CEX_WARNING` key was modified

## Related Issue

Resolve #7377

## Screenshots:

image

image

image

### CODE REVIEW

**Comment by andrewkozlik** [comment]
I think the `PlebStopThreshold` is not published in the coordinator's status. It's a constant in the WasabiWallet config file I think. So that's just for reference. We need to have a similar constant in our codebase and I'd call it `UneconomicalCoinjoinThreshold` instead, ... which you do, so that's good :+1:

File: packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        align-self: start;
+    }
+`;
+
+const Text = styled.div`
+    color: ${({ theme }) => theme.TYPE_DARK_GREY};
+    font-size: ${variables.FONT_SIZE.SMALL};
+`;
+
+const Heading = styled.p`
```
Comment (komret) [inline packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx]:
```suggestion
const Heading = styled.div`
```

See console warning: `validateDOMNesting(...): <div> cannot appear as a descendant of <p>`

File: packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    font-weight: ${variables.FONT_WEIGHT.DEMI_BOLD};
+    text-transform: uppercase;
+`;
+
+const WarningList = styled.ul`
+    padding-left: 16px;
+
+    > li + li {
+        margin-top: 5px;
+    }
```
Comment (komret) [inline packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx]: Nicer IMO:

```suggestion
    display: flex;
    flex-direction: column;
    gap: 5px;
```

File: packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import { selectSelectedAccount } from '@wallet-reducers/selectedAccountReducer';
+import { selectIsAccountWithRatesByKey } from '@suite-common/wallet-core';
+
+const StyledWarning = styled(Warning)`
+    justify-content: space-between;
+    margin-bottom: 16px;
+
+    ${variables.SCREEN_QUERY.MOBILE} {
+        display: grid;
+        grid-template-columns: 1fr auto;
```
Comment (komret) [inline packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx]: Not sure what is the point of `grid-template-columns` here. Anyway, maybe a simpler solution (would not require lines 67-70):
```suggestion
        align-items: stretch;
        flex-direction: column;
        gap: 16px;
```

File: packages/suite/src/components/suite/modals/UnecoCoinjoinWarning.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    const { symbol } = account;
+    const decimals = getAccountDecimals(symbol) || 8;
+
+    const handleContinue = () => {
+        dispatch(onCancel());
+        dispatch(goto('wallet-anonymize', { preserveParams: true }));
+    };
+
+    return (
+        <StyledModal
```
Comment (komret) [inline packages/suite/src/components/suite/modals/UnecoCoinjoinWarning.tsx:66]: Consider adding `isCancelable`.

File: packages/suite/src/components/suite/modals/UnecoCoinjoinWarning.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    font-weight: ${variables.FONT_WEIGHT.DEMI_BOLD};
+`;
+
+const Explanation = styled.i`
+    color: ${({ theme }) => theme.TYPE_LIGHT_GREY};
+    font-size: ${variables.FONT_SIZE.SMALL};
+`;
+
+const CancelButton = styled(Button)`
+    width: 48%;
```
Comment (komret) [inline packages/suite/src/components/suite/modals/UnecoCoinjoinWarning.tsx]: I am not a fan of arbitrary numbers like this, and the button will not adjust if the text is longer. I suggest styling `StyledModal.BottomBar` instead, this time using grid:

```
display: grid;
grid-template-columns: 1fr 1fr;
```

File: packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatusWheel/ProgressWheel.tsx [hunk 6]
```diff
@@ -176,23 +185,32 @@ export const ProgressWheel = ({ accountKey, togglePause }: ProgressWheelProps) =
         isSessionActive,
         togglePause,
         dispatch,
+        isCoinjoinUneco,
     ]);
 
+    const getTooltipMessage = () => {
+        if (isAccountEmpty) {
```
Comment (komret) [inline packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatusWheel/ProgressWheel.tsx:192]: This case is already included in `coinjoinSessionBlockedMessage` variants. The implementation in `coinjoinSessionBlockedMessage` is more robust as it takes into account UTXOs too small for coinjoin.

File: packages/suite/src/support/messages.ts [hunk 7]
```diff
@@ -8065,4 +8061,39 @@ export default defineMessages({
...
+        id: 'TR_UNECO_COINJOIN_STATUS_TOOLTIP',
+        defaultMessage: 'Amount below recommended minimum.',
+    },
+    TR_UNECO_COINJOIN_AGREE: {
+        id: 'TR_UNECO_COINJOIN_AGREE',
+        defaultMessage: 'I understand',
+    },
+    TR_UNECO_COINJOIN_TITLE: {
+        id: 'TR_UNECO_COINJOIN_TITLE',
+        defaultMessage: 'Uneconomical CoinJoin',
```
Comment (komret) [inline packages/suite/src/support/messages.ts]:
```suggestion
        defaultMessage: 'Uneconomical coinjoin',
```

**Review by komret — CHANGES_REQUESTED** [review]
We discussed the UI with @hynek-jina and figured the grey ProgressWheel is a bit misleading as it looks like it is disabled. Could you please keep it in the same state as normal (removing the tooltip, `statusMessage` and making it green)? It should be enough to communicate the message in the modal.

File: packages/suite/src/support/messages.ts (hunk 7 above)
Comment (dahaca) [reply packages/suite/src/support/messages.ts]: 0f15f72eb

File: packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatusWheel/ProgressWheel.tsx (hunk 6 above)
Comment (dahaca) [reply packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatusWheel/ProgressWheel.tsx:192]: Then i'd say the translation for that case should be changed. `Amounts are too small for coinjoin` implies you have _some_ amounts. While `Nothing to anonymize` is explicit in the fact that there is nothing on the account.

File: packages/suite/src/components/suite/modals/UnecoCoinjoinWarning.tsx (hunk 5 above)
Comment (dahaca) [reply packages/suite/src/components/suite/modals/UnecoCoinjoinWarning.tsx]: 0f15f72eb

File: packages/suite/src/components/suite/modals/UnecoCoinjoinWarning.tsx (hunk 4 above)
Comment (dahaca) [reply packages/suite/src/components/suite/modals/UnecoCoinjoinWarning.tsx:66]: yeah I wasn't sure about it, decided to omit. But no strong opinion here :D 

0f15f72eb

File: packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx (hunk 3 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx]: 0f15f72eb

File: packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx]: 0f15f72eb

File: packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/receive/components/CoinjoinReceiveWarning.tsx]: 0f15f72eb

File: packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatusWheel/ProgressWheel.tsx (hunk 6 above)
Comment (komret) [reply packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatusWheel/ProgressWheel.tsx:192]: Ok, you are right.

**Comment by dahaca** [comment]
@komret ec875c99d

**Review by komret — APPROVED** [review]
💯

File: packages/suite/src/actions/suite/storageActions.ts [hunk 8]
```diff
@@ -67,7 +67,7 @@ export const saveCoinjoinAccount =
 const removeCoinjoinRelatedSetting = (state: AppState) => {
     const settings = { ...state.suite.settings };
 
-    settings.isCoinjoinCexWarningHidden = false;
+    settings.isCoinjoinReceiveWarningHidden = false;
```
Comment (matejkriz) [inline packages/suite/src/actions/suite/storageActions.ts:70]: This deserves migration, I guess.

Comment (dahaca) [reply packages/suite/src/actions/suite/storageActions.ts:70]: I was thinking about it but felt too granular to me. A very small amount of users who have their CJ account remembered would have to dismiss it again but that's it. If you think it's needed I will add a migration :)

Comment (matejkriz) [reply packages/suite/src/actions/suite/storageActions.ts:70]: but the value `isCoinjoinCexWarningHidden` will remain in their storage until reset, but ok, maybe it's fine in this case and not needed to add the migration

**Pull request merged**